### PR TITLE
Add table of issue type info and relevant column name descriptions

### PR DESCRIPTION
**Pull request by elisno**
Addresses #1081 

This PR improves the Datalab Issue Types guide.

It adds a table for the different tasks that Datalab supports. It lists out the
- Names of the issues it can find.
- Whether it is searched by default (otherwise it's accessible via the `issue_types` argument in `Datalab.find_issues()`.
- A collection of relevant columns in the `Datalab.issues` dataframe.
- A column that lists out what types of inputs to  `Datalab.find_issues()` are required to successfully run the issue check.

Here's a screenshot of the table (still WIP):
image

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -46,6 +46,7 @@
     "sphinx.ext.napoleon",
     "nbsphinx",
     "sphinx.ext.autodoc",
+    "sphinx.ext.autosectionlabel",
```
Comment (elisno) [inline docs/source/conf.py:49]: This is added to allow us to link to local section headings.

File: docs/source/cleanlab/datalab/guide/issue_type_description.rst [hunk 2]
```diff
@@ -11,6 +11,9 @@ For each type of issue, we explain: what it says about your data if detected, wh
 In case you didn't know: you can alternatively use `Cleanlab Studio <https://cleanlab.ai/blog/data-centric-ai/>`_ to detect the same data issues as this package, plus `many more types of issues <https://help.cleanlab.ai/guide/concepts/cleanlab_columns/>`_, all without having to do any Machine Learning (or even write any code).
 
 
+.. include:: table.rst
+
+
```
[the remark this answers is not in the document]
Comment (jwmueller) [reply docs/source/cleanlab/datalab/guide/issue_type_description.rst:16]: let's add the table, it's awesome!

File: docs/source/cleanlab/datalab/guide/issue_type_description.rst [hunk 3]
```diff
@@ -82,6 +159,39 @@ Closely inspect them and consider removing some outliers that may be negatively
...
+A boolean column, where `True` indicates that an example is identified as an outlier.
+
+``outlier_score``
+~~~~~~~~~~~~~~~~~
+
+A numeric column with scores between 0 and 1. 
+A smaller value for an example indicates that it is less common or typical in the dataset, suggesting that it is more likely to be an outlier.
+
+If most of the nearest-neighbors of an example are exact-duplicates, then the outlier score of the example is set to 1.0.
```
Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/issue_type_description.rst:193]:
```suggestion
```

File: docs/source/cleanlab/datalab/guide/issue_type_description.rst [hunk 4]
```diff
@@ -109,6 +219,53 @@ Including near-duplicate examples in a dataset may negatively impact a ML model'
...
+A numeric column with scores between 0 and 1. The lower the score, the more likely the example is to be a near-duplicate of another example in the dataset.
+
+Exact duplicates are assigned a score of 0, while near-duplicates are assigned a score close to 0.
+
+``near_duplicate_sets``
+~~~~~~~~~~~~~~~~~~~~~~~
+
+A column of lists of integers, where each list contains the indices of examples that belong to the same set of near-duplicates (not including the example itself).
+Each set represents a group of examples that are extremely similar to each other, relative to the rest of the dataset.
+The examples in each set may be exactly duplicated or have very similar feature representations.
```
Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/issue_type_description.rst:258]:
```suggestion
A column of lists of integers. The i-th list contains the indices of examples that are considered near-duplicates of example i (not including example i).
```

File: docs/source/cleanlab/datalab/guide/issue_type_description.rst [hunk 5]
```diff
@@ -109,6 +219,53 @@ Including near-duplicate examples in a dataset may negatively impact a ML model'
...
+
+``distance_to_nearest_neighbor``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A numeric column that represents the distance between each example and its nearest neighbor in the dataset.
+The distance is calculated based on the provided `features` or `knn_graph`.
+A smaller distance indicates that the example is more similar to its nearest neighbor.
+Examples that are (near) duplicates have smaller distances to their nearest neighbors compared to other examples in the dataset.
+Exact duplicates ideally have a distance of 0 to their nearest neighbor. However, due to floating point precision, especially when using certain distance metrics like Euclidean distance, this might not always be the case.
```
Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/issue_type_description.rst:267]:
```suggestion
A numeric column that represents the distance between each example and its nearest neighbor in the dataset.
The distance is calculated based on the provided `features` or `knn_graph`, and is directly related to the `near_duplicate_score`.
A smaller distance indicates that the example is more similar to its nearest neighbor in the dataset.
```

File: docs/source/cleanlab/datalab/guide/issue_type_description.rst [hunk 6]
```diff
@@ -56,6 +59,80 @@ To handle mislabeled examples, you can either filter out the data with label iss
...
+
+A numeric column that gives the label quality score for each example.
+The score lies between 0 and 1.
+The lower the score, the less likely the given label is to be correct.
+
+
+``given_label``
+~~~~~~~~~~~~~~~
+
+A column of the actual labels as provided in the dataset.
```
Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/issue_type_description.rst:129]:
```suggestion
A column of the actual labels as provided in the original dataset.
```

File: docs/source/cleanlab/datalab/guide/issue_type_description.rst [hunk 7]
```diff
@@ -148,6 +342,42 @@ In a dataset identified as having class imbalance, the class imbalance quality s
 
 Class imbalance in a dataset can lead to subpar model performance for the under-represented class. Consider collecting more data from the under-represented class, or at least take special care while modeling via techniques like over/under-sampling, SMOTE, asymmetric class weighting, etc.
```
Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/issue_type_description.rst:343]:
```suggestion
Class imbalance in a dataset can lead to subpar model performance for the under-represented class. Consider collecting more data from the under-represented class, or at least take special care while modeling via techniques like over/under-sampling, SMOTE, asymmetric class weighting, etc.

This issue-type is more about the overall dataset vs. individual data points. If severe class imbalance is detected, Datalab will flag the individual data points from the minority class.
```

File: docs/source/cleanlab/datalab/guide/issue_type_description.rst [hunk 8]
```diff
@@ -148,6 +342,42 @@ In a dataset identified as having class imbalance, the class imbalance quality s
...
+    72                     False                   0.28            2
+    75                     False                   0.28            2
+    33                     False                   0.28            2
+    68                     False                   0.28            2
+
+``is_class_imbalance_issue``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A boolean column, where `True` indicates that the dataset suffers class imbalance. All examples belonging to the rarest class are flagged with this issue.
+But the dataset is considered to have class imbalance only if the proportion of examples belonging to the rarest class must be much lower than the inverse of the number of classes in the dataset.
```
Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/issue_type_description.rst:367]:
```suggestion

A boolean column, where `True` indicates which data points belong to the minority class (rarest class) in a classification dataset that exhibits severe class imbalance.  If the dataset is not considered to have severe class imbalance (i.e. proportion of examples in the rarest class is not to small relative to the number of classes in the dataset), then all values will be `False`.

```

File: docs/source/cleanlab/datalab/guide/issue_type_description.rst [hunk 9]
```diff
@@ -148,6 +342,42 @@ In a dataset identified as having class imbalance, the class imbalance quality s
...
+
+``class_imbalance_score``
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A numeric column with scores between 0 and 1.
+Any example belonging to the most under-represented class is assigned a score equal to the proportion of examples in the dataset belonging to that class.
+All other examples are assigned a score of 1.
```
Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/issue_type_description.rst:374]:
```suggestion
A numeric column with scores between 0 and 1.
Any example belonging to the most under-represented class is assigned a score equal to the proportion of examples in the dataset belonging to that class.
All other examples are assigned a score of 1.
All examples sharing the same label also share the same score.
```

File: docs/source/cleanlab/datalab/guide/issue_type_description.rst [hunk 10]
```diff
@@ -148,6 +342,42 @@ In a dataset identified as having class imbalance, the class imbalance quality s
...
+
+``given_label``
+~~~~~~~~~~~~~~~
+
+A column of the actual labels as provided in the dataset.
```
Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/issue_type_description.rst:379]:
```suggestion
A column of the actual labels as provided in the original dataset.
```

File: docs/source/cleanlab/datalab/guide/issue_type_description.rst [hunk 11]
```diff
@@ -198,6 +461,38 @@ equals the average of the individual examples' quality scores.
...
+    72          False         1.0
+    71          False         1.0
+    70          False         1.0
+    69          False         1.0
+
+``is_null_issue``
+~~~~~~~~~~~~~~~~~
+
+A boolean column, where `True` indicates that an example is identified as having null/missing values across all feature columns.
+Even if an example has a single non-null value, it is not flagged with this issue.
```
Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/issue_type_description.rst:486]:
```suggestion
Examples that just have a single non-null value across multiple feature columns are not flagged with this issue.
```

File: docs/source/cleanlab/datalab/guide/issue_type_description.rst [hunk 12]
```diff
@@ -198,6 +461,38 @@ equals the average of the individual examples' quality scores.
...
+
+``null_score``
+~~~~~~~~~~~~~~
+
+A numeric column with scores between 0 and 1. The score represents the proportion of non-null values in the example.
+A lower score indicates that the example has more null/missing values.
+A score of 0 indicates that all feature values in the example are null. These examples are flagged with the null issue.
+A score of 1 indicates that none of the feature values in the example are null.
```
Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/issue_type_description.rst:494]:
```suggestion
A numeric column with scores between 0 and 1. The score represents the proportion of non-null (i.e. non-missing) values in each example.
Lower scores indicate examples with more null/missing values.
```

File: docs/source/cleanlab/datalab/guide/issue_type_description.rst [hunk 13]
```diff
@@ -213,6 +508,36 @@ Data valuation issues can be detected based on provided `features` or a provided
...
+    39                    False                   0.5
+    32                    False                   0.5
+    98                    False                   0.5
+    6                     False                   0.5
+    7                     False                   0.5
+
+``is_data_valuation_issue``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A boolean column, where `True` indicates that an example is contributes negatively to a model's training performance.
```
Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/issue_type_description.rst:532]:
```suggestion
A boolean column, where `True` indicates that an example does not appear to contribute positively to a model's training performance.
```

File: docs/source/cleanlab/datalab/guide/issue_type_description.rst [hunk 14]
```diff
@@ -213,6 +508,36 @@ Data valuation issues can be detected based on provided `features` or a provided
...
+
+``data_valuation_score``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+A numeric column with scores between 0 and 1. The score reflects how valuable an example is in terms of improving or maintaining the performance of the model during training.
+A score below 0.5 indicates that the example contributes negatively to the model's performance. Such examples are flagged as having a data valuation issue.
+A score above 0.5 indicates that the example positively influences the model's performance, contributing to better learning and predictive accuracy.
```
Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/issue_type_description.rst:539]:
```suggestion
A numeric column with scores between 0 and 1. The score reflects how valuable each individual example is in terms of improving the performance of the ML model trained on this dataset.
Examples with higher scores more positively influence the resulting model's predictive performance, contributing to better learning. One would expect the model to get worse if many such examples were removed from its training dataset.
```

**Review by jwmueller — APPROVED** [review]
LGTM! Pushed many changes, so please review my commits closely before you merge this

File: docs/source/cleanlab/datalab/guide/table.rst [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+              - (`pred_probs`, `features`) or (`pred_probs`, `knn_graph`) or (`pred_probs`, `cluster_ids`) [#f2]_
+            * - :ref:`null<Null Issue>`
+              - YES
+              - 
+                :ref:`is_null_issue<\`\`is_null_issue\`\`>`
+                
+                :ref:`null_score<\`\`null_score\`\`>`
+              - `features`
+            * - :ref:`data_valuation<Data Valuation Issue>`
+              - no
```
Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/table.rst:71]: why is this lower case?

Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/table.rst:71]:
```suggestion
              - NO
```

Comment (jwmueller) [reply docs/source/cleanlab/datalab/guide/table.rst:71]: I'd actually suggest make the yes's: `Yes` and this one `No` vs. all caps.
But definitely not some all-caps some not.

File: docs/source/cleanlab/datalab/guide/table.rst [hunk 16]
```diff
@@ -0,0 +1,186 @@
+.. tabs::
+
+    .. tab:: Classification task
```
Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/table.rst:3]: opened another TODO related to this table:
https://github.com/cleanlab/cleanlab/issues/1133

But can save for the future

File: docs/source/cleanlab/datalab/guide/table.rst (hunk 15 above)
Comment (elisno) [reply docs/source/cleanlab/datalab/guide/table.rst]: Capitalized it. Thanks!

Comment (elisno) [reply docs/source/cleanlab/datalab/guide/table.rst]: Done.

File: docs/source/cleanlab/datalab/guide/table.rst (hunk 16 above)
Comment (elisno) [reply docs/source/cleanlab/datalab/guide/table.rst:3]: I agree

**Pull request merged**